airbyte-integrations/connectors/destination-weaviate/destination_weaviate/client.py

```diff
@@ -0,0 +1,97 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import uuid
+import logging
+import json
+from typing import Any, Mapping
+
+import weaviate
+from .utils import generate_id, parse_id_schema, parse_vectors, stream_to_class_name
+
+
+class Client:
+    def __init__(self, config: Mapping[str, Any], schema: Mapping[str, str]):
+        self.client = self.get_weaviate_client(config)
+        self.config = config
+        self.batch_size = int(config.get("batch_size", 100))
+        self.schema = schema
+        self.vectors = parse_vectors(config.get("vectors"))
+        self.id_schema = parse_id_schema(config.get("id_schema"))
+
+    def queue_write_operation(self, stream_name: str, record: Mapping):
+        # TODO need to handle case where original DB ID is not a UUID
+        record_id = ""
```

Comment:
This variable is marked as 'Local variable 'record_id' value is not used' is this some scope or indentation issue?

Reply:
It's used later in this line https://github.com/airbytehq/airbyte/pull/20094/files/a5d8d2330a42a83cbf0b1553696968955da457dc#diff-388d50934c4a187ab902e81c5d858652c09da960fa6ba179f6bdbba7c527980fR59:
```
self.client.batch.add_data_object(record, class_name, record_id, vector=vector)
```

I think it's an issue in the IDE. My IDE shows the same thing even though its used later in the function.

Reply:
seems it's not needed and my python understanding was incorrect: https://stackoverflow.com/questions/2829528/whats-the-scope-of-a-variable-initialized-in-an-if-statement

I've removed line 25 and all still works as expected

Reply:
Python made a pretty weird choice here. being able to reference a variable that was declared in a different scope...